src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```diff
@@ -1,84 +1,185 @@
 
 
 case class NihStatus(
-  loginRequired: Boolean,
   linkedNihUsername: Option[String] = None,
   isDbgapAuthorized: Option[Boolean] = None,
-  lastLinkTime: Option[Long] = None,
   linkExpireTime: Option[Long] = None,
-  descriptionSinceLastLink: Option[String] = None,
   descriptionUntilExpires: Option[String] = None
 )
 
 object NihStatus {
 
-  implicit val impNihStatus = jsonFormat7(NihStatus.apply)
+  implicit val impNihStatus = jsonFormat4(NihStatus.apply)
 
   def apply(profile: Profile): NihStatus = {
     apply(profile, profile.isDbgapAuthorized)
   }
 
   def apply(profile: Profile, isDbGapAuthorized: Option[Boolean]): NihStatus = {
-    val linkExpireSeconds = profile.linkExpireTime.getOrElse(0L)
-    val howSoonExpire = DateUtils.secondsSince(linkExpireSeconds)
     new NihStatus(
-      loginRequired = howSoonExpire >= 0,
       profile.linkedNihUsername,
       isDbGapAuthorized,
-      profile.lastLinkTime,
       profile.linkExpireTime
     )
   }
 }
 
 object NihService {
   sealed trait ServiceMessage
-  case class GetStatus(userInfo: UserInfo) extends ServiceMessage
+  case class GetNihStatus(userInfo: UserInfo) extends ServiceMessage
+  case class UpdateNIHLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink) extends ServiceMessage
+  case object SyncWhitelist extends ServiceMessage
 
   def props(service: () => NihService): Props = {
     Props(service())
   }
 
   def constructor(app: Application)()(implicit executionContext: ExecutionContext) =
-    new NihService(app.rawlsDAO, app.thurloeDAO)
+    new NihService(app.rawlsDAO, app.thurloeDAO, app.googleServicesDAO)
 }
 
-class NihService(val rawlsDao: RawlsDAO, val thurloeDao: ThurloeDAO)
+class NihService(val rawlsDao: RawlsDAO, val thurloeDao: ThurloeDAO, val googleDao: GoogleServicesDAO)
   (implicit protected val executionContext: ExecutionContext) extends Actor
-  with LazyLogging {
+  with LazyLogging with FireCloudRequestBuilding {
 
   override def receive = {
-    case GetStatus(userInfo: UserInfo) => getStatus(userInfo) pipeTo sender
+    case GetNihStatus(userInfo: UserInfo) => getNihStatus(userInfo) pipeTo sender
+    case UpdateNIHLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink) => updateNihLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink) pipeTo sender
+    case SyncWhitelist => syncWhitelist() pipeTo sender
   }
 
-  private def getStatus(userInfo: UserInfo): Future[PerRequestMessage] = {
+  private def getNihStatus(userInfo: UserInfo): Future[PerRequestMessage] = {
     thurloeDao.getProfile(userInfo) flatMap {
       case Some(profile) =>
         profile.linkedNihUsername match {
           case Some(_) =>
-            rawlsDao.isDbGapAuthorized(userInfo) map { x =>
-              RequestComplete(NihStatus(profile, Some(x)))
+            rawlsDao.isDbGapAuthorized(userInfo) map { isDbGapAuthorized =>
+              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), Some(isDbGapAuthorized)))
```

Comment:
if the linkExpireTime is missing from their profile, we return their expire time as the beginning of time so they'll be prompted to re-link in the UI. they will also be removed once the whitelist syncs again, but that's separate logic